### PR TITLE
Fix: stack trace of exceptions are not being logged #1974

### PR DESCRIPTION
**Pull request by nitram509**
### Proposed changes in this pull request

This is a fix for #1974 Stack trace of exceptions are not being logged.

* I did some regex search and replaced all ca. 50 occurrences I found, where " + e" was used instead of ", e" when logging.
   * Used regex expressions: '\Wlog\.\(.*\s\+\se' and simply '\s\+\se'
* I also added some ", e" parameters to Exceptions and Log statements, where I thought it might be useful.
* Additionally I fixed a test, where some error logging will not make the test fail, but a RuntimeException is preferred to fail fast. 
* Also, there are some trailing whitespaces removed (this is automatic by my IDE), I do apologize for this noise. If you don't like it, I can move them back in.

### When should this PR be merged

I don't see any pre-conditions.
Could be merged right away.

### Follow up actions

* a review would be nice
* feedback would be nice
* setting a label "hacktoberfest-accepted" would be awesome.

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

**Comment by nitram509** [comment]
Dear reviewer,
I realized my first push did reveal an overseen Exeception thrown in the test class FunctionLibraryManagementServiceTest.
After reading the code, I concluded, there must be a mistake in the data provider setup, for "sampleFunction3".
Hence I fixed it.
Also, I did replace all log statements with Asster.fail() code, because this would make a test fail instead of silently logging.

I hope that's in your interest.

Any feedback welcome.

File: components/functions-library-mgt/org.wso2.carbon.identity.functions.library.mgt/src/test/java/org/wso2/carbon/identity/functions/library/mgt/FunctionLibraryManagementServiceTest.java [hunk 1]
```diff
@@ -34,17 +32,14 @@
 import java.util.List;
 
 import static org.powermock.api.mockito.PowerMockito.when;
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertNotNull;
-import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.*;
```
Comment (pulasthi7) [inline components/functions-library-mgt/org.wso2.carbon.identity.functions.library.mgt/src/test/java/org/wso2/carbon/identity/functions/library/mgt/FunctionLibraryManagementServiceTest.java:35]: As a practice, we avoid * imports. So, please revert this back to individual imports

**Review by pulasthi7 — COMMENTED** [review]
Hi @nitram509 

Thanks for your contribution. Much Appreciated. 

I'm adding one minor comment to be fixed. Please get that addressed and the PR would be ready to be merged.

File: components/functions-library-mgt/org.wso2.carbon.identity.functions.library.mgt/src/test/java/org/wso2/carbon/identity/functions/library/mgt/FunctionLibraryManagementServiceTest.java (hunk 1 above)
Comment (nitram509) [reply components/functions-library-mgt/org.wso2.carbon.identity.functions.library.mgt/src/test/java/org/wso2/carbon/identity/functions/library/mgt/FunctionLibraryManagementServiceTest.java:35]: 👍

Comment (nitram509) [reply components/functions-library-mgt/org.wso2.carbon.identity.functions.library.mgt/src/test/java/org/wso2/carbon/identity/functions/library/mgt/FunctionLibraryManagementServiceTest.java]: @pulasthi7 Thank you for your review.
I did change the import as per convention.

**Review by pulasthi7 — APPROVED** [review]

File: components/functions-library-mgt/org.wso2.carbon.identity.functions.library.mgt/src/test/java/org/wso2/carbon/identity/functions/library/mgt/FunctionLibraryManagementServiceTest.java (hunk 1 above)
Comment (pulasthi7) [reply components/functions-library-mgt/org.wso2.carbon.identity.functions.library.mgt/src/test/java/org/wso2/carbon/identity/functions/library/mgt/FunctionLibraryManagementServiceTest.java]: Thanks @nitram509

**Pull request merged**